sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java

```diff
@@ -909,14 +856,14 @@ public DataLakeServiceVersion getServiceVersion() {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<AccessControlChangeResult>> setAccessControlRecursiveWithResponse(
         PathSetAccessControlRecursiveOptions options) {
-        StorageImplUtils.assertNotNull("options", options);
         try {
+            StorageImplUtils.assertNotNull("options", options);
```

Comment:
is this ok to do ? I.e. emit mono instead of throwing right away?

Reply:
Are you asking if throwing is the correct behavior or returning a Mono error? Either way, this retains previous behavior but centralizes the logic to this API. Before the async caller wrapped the thrown exception in a Mono error and the sync caller will block right away trigger Reactor to throw the exception on call